awx/main/tests/live/tests/test_cleanup_task.py

```diff
@@ -37,3 +47,34 @@ def delete_jobs():
     print('ansible-runner worker ' + remote_command)
 
     assert [os.path.exists(job_dir) for job_dir in job_dirs] == [True for i in range(3)]
+
+
+@pytest.mark.parametrize(
+    'worktype',
+    ('remote', 'local'),
+)
+def test_tagless_image(podman_image_generator, worktype):
+    """
+    Ensure podman images on Control and Hybrid nodes are deleted during cleanup.
+    """
+    podman_image_generator()
+
+    dangling_image = next((image for image in get_podman_images() if image.get('Dangling', False)), None)
+    assert dangling_image
+
+    with (
+        (
+            mock.patch('awx.main.tasks.receptor.run_until_complete', lambda *args, **kwargs: run_until_complete(*args, worktype='local', ttl=None, **kwargs))
+            if worktype == 'local'
+            else nullcontext()
+        ),
+        (
+            mock.patch('awx.main.tasks.system.CleanupImagesAndFilesHelper.get_execution_instances', lambda: [Instance.objects.me()])
+            if worktype == 'local'
+            else nullcontext()
+        ),
+    ):
+        _cleanup_images_and_files(image_prune=True)
```

Comment:
My only nit pick about this test is now that this call with run _both_ the control & execution forms of the logic. Meaning it will run `ansible_runner.cleanup.run_cleanup(runner_cleanup_kwargs)` before the command over receptor. So if you're testing that the CLI args do not error, this test is sufficient. But it doesn't test that the command ran over receptor _actually cleans up the images_ because this would have already been done in the control form of the logic. To address this, I am pretty sure you'd have to break up `_cleanup_images_and_files` into 2 methods or something like that. I think that would be better for cleanness and also better testing. But as for the first goal of regression testing that we don't get errors from the execution node command, this is doing the job already.